src/commons/body-message-renderer.jsx

```diff
@@ -54,34 +56,23 @@ function TextMessageRenderer({ text }) {
 
 function HtmlMessageRenderer({ msgId, body, parts }) {
 	const iframeRef = useRef();
-	const onIframeLoad = useCallback((ev) => {
-		ev.persist();
-		const styleTag = document.createElement('style');
-		const styles = `
-			body {
-				margin: 0;
-				overflow-y: hidden;
-				font-family: Roboto, sans-serif;
-				font-size: 0.875rem;
-			}
-			body pre, body pre * {
-				white-space: pre-wrap;
-				word-wrap: break-word !important;
-				text-wrap: suppress !important;
-			}
-			img {
-				max-width: 100%
-			}
-			
-		`;
-		styleTag.textContent = styles;
-		iframeRef.current.contentDocument.head.append(styleTag);
-		iframeRef.current.style.display = 'block';
-		iframeRef.current.style.height = `${
-			iframeRef.current.contentDocument.body.querySelector('div').scrollHeight
-		}px`;
+
+	const convertInRem = useCallback((px, base = 16) => {
+		let tempPx = px;
+		if (typeof px === 'string' || px instanceof String) tempPx = tempPx.replace('px', '');
+
+		tempPx = parseInt(tempPx, 10);
+		return `${(1 / base) * tempPx}rem`;
 	}, []);
 
+	const calculateHeight = useCallback(() => {
+		if (!isNull(iframeRef.current)) {
+			const scrollHeight = iframeRef.current.contentDocument.querySelector('html')?.scrollHeight;
+			iframeRef.current.style.height = '0';
+			iframeRef.current.style.height = convertInRem(scrollHeight ?? '0');
+		}
+	}, [convertInRem]);
+
 	const updatedBody = useMemo(() => replaceLinkToAnchor(body), [body]);
 
 	useEffect(() => {
```

Comment:
did you already consider a useLayoutEffect? it seems to me the most appropriate use case

Reply:
I updated it and also updated the stle using the one used in mails